ComplementaryScripts/modelCuration/addBiologRxnToGEM.m

```diff
@@ -0,0 +1,170 @@
+end
+
+% Map mets to model.metnames, get s_index for new mets:
+cd otherChanges
+for j = 1:length(matrix.metnames)
+    [~,metindex] = ismember(matrix.metnames(j),model.metNames);
+    if metindex ~= 0
+        matrix.mets(j) = model.mets(metindex);
+    elseif metindex == 0
+        newID = getNewIndex(model.mets);
+        matrix.mets(j) = strcat('s_',newID,matrix.Newcomps(j));
+        model = addMetabolite(model,char(matrix.mets(j)), ...
+            'metName',matrix.metnames(j));
+    end
+end
+
+% Add metabolite data:
+fid = fopen('../../ComplementaryData/modelCuration/Biolog_newRxnMetAnnotation.tsv');
+newmet_annot         = textscan(fid,'%s %s %s %s %s %s %s','Delimiter','\t','HeaderLines',1);
+newmet.metNames      = newmet_annot{1};
+newmet.metFormulas   = newmet_annot{2};
+newmet.metCharges    = cellfun(@str2num, newmet_annot{3});
+newmet.metKEGGID     = newmet_annot{5};
+newmet.metChEBIID    = newmet_annot{6};
+newmet.metMetaNetXID = newmet_annot{7};
+
+fclose(fid);
+for i = 1:length(newmet.metNames)
+    [~,metID] = ismember(newmet.metNames(i),model.metNames);
+    if metID ~= 0
+        model.metFormulas{metID}   = newmet.metFormulas{i};
+        model.metCharges(metID)    = newmet.metCharges(i);
+        model.metKEGGID{metID}     = newmet.metKEGGID{i};
+        model.metChEBIID{metID}    = newmet.metChEBIID{i};
+        model.metMetaNetXID{metID} = newmet.metMetaNetXID{i};
+        model.metNotes{metID}      = '';
+    end
+end
+
+% Add new reactions according to rev ID: Met Coef needs to be a column, not
+% a row. Coef should be a double, which was converted at the import section
+EnergyResults     = {};
+MassChargeresults = {};
+RedoxResults      = {};
+if ~isfield(model,'rxnMetaNetXID')
+    model.rxnMetaNetXID = cell(size(model.rxns));
+end
+for i = 1:length(newrxn.ID)
+    newID = getNewIndex(model.rxns);
+    j     = find(strcmp(matrix.rxnIDs,newrxn.ID{i}));
+    Met   = matrix.mets(j);
+    Coef  = transpose(matrix.metcoef(j));
+    [model,rxnIndex] = addReaction(model, ['r_' newID],...
+        'reactionName', newrxn.ID{i},...
+        'metaboliteList',Met,...
+        'stoichCoeffList',Coef,...
+        'reversible',newrxn.Rev(i,1),...
+        'geneRule',newrxn.GPR{i},...
+        'checkDuplicate',1);
+    [EnergyResults,RedoxResults] = CheckEnergyProduction(model,{['r_' newID]},EnergyResults,RedoxResults);
+    [MassChargeresults] = CheckBalanceforSce(model,{['r_' newID]},MassChargeresults);
+    if isempty(rxnIndex)
+        rxnIndex = strcmp(model.rxns,['r_' newID]);
+    end
+    % Add rxn annotation:
+    model.rxnNames{rxnIndex}      = newrxn.rxnNames{i};
+    model.rxnECNumbers(rxnIndex)  = newrxn.rxnECNumbers(i);
+    model.rxnKEGGID(rxnIndex)     = newrxn.rxnKEGGID(i);
+    model.rxnMetaNetXID(rxnIndex) = newrxn.rxnMetaNetXID(i);
+    model.rxnConfidenceScores(rxnIndex) = 1;   %reactions without gene but needed for modelling
+end
+
+% Add gene standard name for new genes:
```

Comment:
Yes, we can delete this part.

Reply:
but that section does not have to do with GPRs, it adds gene names, and no gene is added...

Reply:
One rxn that we added here has gene rules, even though we don't add new genes, but in case that we add new genes later, so maybe it is better to keep that?

Reply:
I think it's misleading: adding new genes should be reserved for branches that add new annotation, like PRs #124 and #142, and should not be done here. On a side note, why did the following 3 rxns included gene rules? Should they not have been added in PR #142? 
```
( YHR201C or YDR452W )			Triphosphate phosphohydrolase
( YIL053W or YER062C )			glycerol-1-phosphate phosphohydrolase
( YJR130C or YLL058W or YML082W )	O4-succinyl-L-homoserine:hydrogen sulfide S-(3-amino-3-carboxypropyl)transferase; O-succinyl-L-homoserine succinate-lyase (adding hydrogen sulfide)
```

Reply:
It is because in the PR #142, we added new genes and new rxns related to those new genes, and we checked whether the relation between existing genes and existing rxns are correct, but we didn't check the new rxns related to the existing genes.

Reply:
@feiranl I see. Should we not include them though? If so I can start an issue for it.

Reply:
Okay, then we remove that to let the branch exactly for Biolog update. It's better to start an issue for it, then we can fix that later.

Reply:
Also, as all new genes have been already added in #142, I will delete the following section of this function, to avoid confusion:
```matlab
% Add gene standard name for new genes:
fid = fopen('../../ComplementaryData/databases/SGDgeneNames.tsv');
yeast_gene_annotation = textscan(fid,'%s %s','Delimiter','\t','HeaderLines',1);
fclose(fid);
geneIndex = zeros(1,1);
for i = 1: length(model.genes)
    geneIndex = strcmp(yeast_gene_annotation{1}, model.genes{i});
    if sum(geneIndex) == 1 && ~isempty(yeast_gene_annotation{2}{geneIndex})
        model.geneNames{i} = yeast_gene_annotation{2}{geneIndex};
    else
        model.geneNames{i} = model.genes{i};
    end
end
% Add protein name for genes:
for i = 1:length(model.genes)
    model.proteins{i} = strcat('COBRAProtein',num2str(i));
end
```